### PR TITLE
Improved Connection Handlings, Error-Message clarification + Channel Selection

### PR DESCRIPTION
**Pull request by JulianMa**
The Post-Init will now wait, until a connection to discord is established and the guild data is downloaded.
Also:
* Improved Error-Messages for Channel Permissions and server id not found
* Sort Channel-List for GUI Channel Picker + Null-Handling in Disconnected State

### CODE REVIEW

File: DiscordLink/Source/Data/DiscordChannelNameTypeConverter.cs [hunk 1]
```diff
@@ -69,6 +74,7 @@ public override object ConvertTo(ITypeDescriptorContext context, CultureInfo cul
         {
             if (value != null && destinationType == typeof(string) && value.GetType() == typeof(ulong))
             {
+                if (DiscordLink.Obj?.Client?.Guild?.Channels?.Values == null) return "No Connection";
```
Comment (MonzUn) [inline DiscordLink/Source/Data/DiscordChannelNameTypeConverter.cs]: Should have pointed this out in the previous merge, but the style I've used is to not do one line ifs.
Could you swap these to this style?
if()
   return

File: DiscordLink/Source/Core/DiscordLink.cs [hunk 2]
```diff
@@ -144,6 +151,32 @@ private async void PostServerInitialize()
...
+            int connectingTimePassed = 0;
+            while (Client.ConnectionStatus == DLDiscordClient.ConnectionState.Connecting)
+            {
+                connectingTimePassed++;
+                await Task.Delay(1000);
+
+                // Warn after 5 seconds, then every minute.
+                if (connectingTimePassed == 5 || connectingTimePassed % 60 == 0)
+                {
+                    Logger.Info(
```
Comment (MonzUn) [inline DiscordLink/Source/Core/DiscordLink.cs]: Is this linebreak intended?

File: DiscordLink/Source/Core/DiscordLink.cs [hunk 3]
```diff
@@ -120,19 +121,25 @@ public void Initialize(TimedTask timer)
 
         private async void PostServerInitialize()
         {
-            Status = "Performing post server start initialization";
+            Status = "Awaiting Guild Download..";
```
Comment (MonzUn) [inline DiscordLink/Source/Core/DiscordLink.cs]: Should this be "..."?

**Review by MonzUn — COMMENTED** [review]
Great fixes! Thank you! :)

File: DiscordLink/Source/Core/DiscordLink.cs (hunk 2 above)
Comment (JulianMa) [reply DiscordLink/Source/Core/DiscordLink.cs]: yeah, its my formatter.. Will change it

File: DiscordLink/Source/Core/DiscordLink.cs (hunk 3 above)
Comment (JulianMa) [reply DiscordLink/Source/Core/DiscordLink.cs]: Ah, no, that doesnt make sense here :D

**Pull request merged**